src/OrleansProviders/Streams/Common/Monitors/ICacheMonitor.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Orleans.Providers.Streams.Common
+{
+    /// <summary>
+    /// Responsible for monitoring cache related metrics
+    /// </summary>
+    public interface ICacheMonitor
+    {
+        #region Event driven metrics
+
+        /// <summary>
+        /// Track cache pressure metrics when cache pressure monitor encounter a status change
+        /// </summary>
+        void TrackCachePressureMonitorStatusChange(string pressureMonitorType, bool underPressure, double? cachePressureContributionCount, double? currentPressure);
```

Comment:
I understand.  Threshold is a configuration, not a metric, however it can be important to mentoring, as I monitoring, it can influence severity, and the only means a service developer would have of getting at that info, should we not pass it in, would be to map the pressureMonitorType to the threshold.

These monitoring hooks expose metrics, but they will likely be used for both metric reporting and critical logging, especially for alerting purposes.

As for the user knowing these values, a developer intimate with the system may well, but a maintainer looking at logs will likely not.  Pls keep in mind some services use many different stream providers, possibly with many different thresholds, so the likelihood of an engineer knowing these values may be low.